### PR TITLE
rename serverFetch to externalFetch

### PR DESCRIPTION
**Pull request by Mlocik97**
Closes https://github.com/sveltejs/kit/issues/2101

---

### Before submitting the PR, please make sure you do the following
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/src/core/dev/index.js [hunk 1]
```diff
@@ -346,7 +346,7 @@ async function create_handler(vite, config, dir, cwd, manifest) {
 						hooks: {
 							getSession: hooks.getSession || (() => ({})),
 							handle: hooks.handle || (({ request, resolve }) => resolve(request)),
-							serverFetch: hooks.serverFetch || fetch
+							externalFetch: hooks.externalFetch || fetch
```
Comment (benmccann) [inline packages/kit/src/core/dev/index.js:349]: copying a comment over from the old PR

on line ~293 we have a check for old hooks. we should add a check for `serverFetch` as well

```
				if (/** @type {any} */ (hooks).getContext) {
					// TODO remove this for 1.0
					throw new Error(
						'The getContext hook has been removed. See https://kit.svelte.dev/docs#hooks'
					);
				}
```

Comment (Mlocik97) [reply packages/kit/src/core/dev/index.js:349]:
```js
if (/** @type {any} */ (serverFetch).getContext) {
					// TODO remove this for 1.0
					throw new Error(
						'The serverFetch was renamed to externalFetch. See https://kit.svelte.dev/docs#hooks-externalfetch'
					);
				}
```

so something like this?

Comment (Mlocik97) [reply packages/kit/src/core/dev/index.js:349]: I tried, but I'm totally lost in this. I don't know how to define value of if condition.

Comment (benmccann) [reply packages/kit/src/core/dev/index.js:349]: I would think that on line 300, you can add:
```
				if (/** @type {any} */ (hooks).serverFetch) {
					// TODO remove this for 1.0
					throw new Error(
						'The serverFetch hook has been renamed to externalFetch'
					);
				}
```

Comment (Mlocik97) [reply packages/kit/src/core/dev/index.js:349]: oh, ofc... I'm dumb.

**Review by benmccann — APPROVED** [review]

**Comment by benmccann** [comment]
oh, it looks like the Lint job is failing. can you fix that?

**Comment by Mlocik97** [comment]
oh, I forgot to run `pnpm format`, should be fine now.

**Comment by stalkerg** [comment]
ok, looks fine for me

File: packages/kit/types/index.d.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ export { ErrorLoad, ErrorLoadInput, Load, LoadInput, LoadOutput, Page } from './
 export {
 	GetSession,
 	Handle,
-	ServerFetch,
+	ExternalFetch,
```
Comment (ignatiusmb) [inline packages/kit/types/index.d.ts:12]: Please put this before `GetSession` for sorting consistency

File: packages/kit/types/internal.d.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { RequestHandler } from './endpoint';
 import { Headers, Location, ParameterizedBody } from './helper';
-import { GetSession, Handle, ServerResponse, ServerFetch, StrictBody } from './hooks';
+import { GetSession, Handle, ServerResponse, ExternalFetch, StrictBody } from './hooks';
```
Comment (ignatiusmb) [inline packages/kit/types/internal.d.ts:3]: Same as above

File: documentation/docs/04-hooks.md [hunk 4]
```diff
@@ -92,19 +92,19 @@ export function getSession(request) {
 
 > `session` must be serializable, which means it must not contain things like functions or custom classes, just built-in JavaScript data types
 
-### serverFetch
+### externalFetch
 
 This function allows you to modify (or replace) a `fetch` request for an **external resource** that happens inside a `load` function that runs on the server (or during pre-rendering).
```
Comment (ignatiusmb) [inline documentation/docs/04-hooks.md:97]: We should also update this while we're at it

> That the docs need to say in bold that it's external is a bit of a code smell.

**Comment by Kapsonfire-DE** [comment]
@benmccann @stalkerg 
I guess we should rename the function before 1.0 and make externalFetch an alias with deprecation warning
as early people change their code the less makes it breaking

**Comment by benmccann** [comment]
I think the current check for old `serverFetch` name should be sufficient. The first time someone runs their app they'll be told to update their code. I actually think this is safer in a lot of ways

**Comment by Rich-Harris** [comment]
thank you!

**Pull request merged**